comment/validators.py

```diff
@@ -135,3 +140,37 @@ def is_valid(self):
             return True
         except ValidationError:
             return False
+
+
+@lru_cache()
+def _get_allowed_orders():
+    allowed_orders = ['reaction__likes', 'reaction__dislikes', 'posted']
+    # adds support for descending order as well
+    allowed_orders.extend(list(map(lambda a: '-' + a, allowed_orders)))
+    # django allows this to support random order
+    allowed_orders.append('?')
+    return allowed_orders
+
+
+def _validate_orders_are_unique(preferred_orders):
+    unique_values = set(map(lambda a: a.replace('-', ''), preferred_orders))
+    if len(unique_values) != len(preferred_orders):
+        duplicated_orders = list(set(preferred_orders) - unique_values)
+        raise ImproperlyConfigured(
+            ErrorMessage.DUPLICATE_ORDER_VALUE.format(
+                duplicates=duplicated_orders, order=duplicated_orders[0].replace('-', '')
+            )
+        )
+
+
+def _validate_order():
```

Comment:
Any reason for making/tagging this function as private `_validate_order` despite it's intended to be used outside the module?

Reply:
internal functions too can be prefixed with  `_`. Just to remind people that this method may not be backward compatible in the future. I think we have gone over this stuff in the past  as well.

Reply:
I am not sure this fits to our case. Our app is just a plug-in not a framework.

Reply:
Since we are exposing our code to the client, this is just a way to tell them not to rely on this function. It is for internal use and any changes can be made in into it without a major version bump.

Reply:
Where did you get this info from?
Can you please share this info?
I personally didn't see frameworks prefix their methods and functions with `_` to let people know that it is not backward compatible, otherwise we will see frameworks full of prefixed `_` functions.
Also this is developers invention not a language built-in and it's evantually up to the user's preference and as my preference I don't like to use them. :)

Reply:
> Where did you get this info from?
Can you please share this info?

I'm also quoting the relevant sections here.

Python: https://docs.python.org/3/tutorial/classes.html#tut-private
>> “Private” instance variables that cannot be accessed except from inside an object don’t exist in Python. However, there is a convention that is followed by most Python code: a name prefixed with an underscore (e.g. _spam) should be treated as a non-public part of the API (whether it is a function, a method or a data member). It should be considered an implementation detail and subject to change without notice.

Django: https://docs.djangoproject.com/en/dev/misc/api-stability/#apis-marked-as-internal

>> Certain APIs are explicitly marked as “internal” in a couple of ways:
    Some documentation refers to internals and mentions them as such. If the documentation says that something is internal, we reserve the right to change it.
    Functions, methods, and other objects prefixed by a leading underscore (_). This is the standard Python way of indicating that something is private; if any method starts with a single _, it’s an internal API.

If I remember correctly, we have gone over this earlier, when there was a discussion around using ` _get_version` with a leading `_` inside the `__init__` module. 


> Also this is developers invention not a language built-in and it's evantually up to the user's preference and as my preference I don't like to use them. :)

Yes, this is developer invention and so, is the complete language. I like this convention because it explicitly tells the users about the stability of the API.

Reply:
Cool, Thanks for sharing. So let's use this convention.